### ISSUE TITLE
`unnecessary_parenthesis` false positive with null-aware + cascade + assignment

### ISSUE THREAD

**Issue by oprypin** [dart-lang/linter#4041]
False positive for this case:
```dart
class Foo {
  var bar = false;
  void method() {}
}

void testLint(Foo? foo) {
  (foo?..bar = true)?.method();    // Unnecessary use of parentheses.
}
```

Wrong fix that is being suggested:
```dart
void testLint(Foo? foo) {
  foo?.bar = true.method();
}
```
